toolchain/check/testdata/array/fail_bound_negative.carbon

```diff
@@ -4,27 +4,37 @@
 //
 // AUTOUPDATE
 
-// TODO: Once we supoprt negating integer literals, check that we reject a
-// negative array bound.
+fn Negate(n: i32) -> i32 = "int.negate";
 
-// CHECK:STDERR: fail_bound_negative.carbon:[[@LINE+3]]:14: ERROR: Semantics TODO: `missing or invalid operator interface`.
-// CHECK:STDERR: var a: [i32; -1];
-// CHECK:STDERR:              ^~
-var a: [i32; -1];
+// CHECK:STDERR: fail_bound_negative.carbon:[[@LINE+3]]:14: ERROR: Array bound of -1 is negative.
+// CHECK:STDERR: var a: [i32; Negate(1)];
+// CHECK:STDERR:              ^~~~~~~
+var a: [i32; Negate(1)];
 
 // CHECK:STDOUT: --- fail_bound_negative.carbon
 // CHECK:STDOUT:
 // CHECK:STDOUT: constants {
 // CHECK:STDOUT:   %.1: i32 = int_literal 1 [template]
+// CHECK:STDOUT:   %.2: i32 = int_literal 4294967295 [template]
```

Comment:
Looks like we don't handle negative integers here.

Reply:
Sent #3822 to fix this.